Source/BestProjectEver/Private/Weapons/BPE_Weapon.cpp

```diff
@@ -238,37 +247,48 @@ void ABPE_Weapon::Fire()
 				break;
 			}
 		}
+		
+		ApplyDamage(HitTarget, ShootDirection);
 	}
 }
 
 //----------------------------------------------------------------------------------------------------------------------
-void ABPE_Weapon::TraceUnderCrosshairs(FHitResult& OutHitResult)
+void ABPE_Weapon::ApplyDamage(const FHitResult& HitResult, FVector ShootDirection)
 {
-	const ABPE_PlayerCharacter* PlayerOwner = Cast<ABPE_PlayerCharacter>(OwnerCharacter);
+	AActor* DamagedActor = HitResult.GetActor();
+	if(IsValid(DamagedActor) && IsValid(OwnerCharacter))
+	{
+		UGameplayStatics::ApplyPointDamage(DamagedActor, BaseDamage, ShootDirection, HitResult,
+			OwnerCharacter->GetInstigatorController(),OwnerCharacter, DamageType);	
+	}
+}
 
-	if(IsValid(PlayerOwner))
+//----------------------------------------------------------------------------------------------------------------------
+void ABPE_Weapon::TraceUnderCrosshairs(FHitResult& OutHitResult, FVector& Direction)
+{
+	if(IsValid(OwnerCharacter))
 	{
 		FVector EyeLocation;
 		FRotator EyeRotation;
-		PlayerOwner->GetActorEyesViewPoint(EyeLocation, EyeRotation);
-
-		const FVector ShotDirection = EyeRotation.Vector();
+		OwnerCharacter->GetActorEyesViewPoint(EyeLocation, EyeRotation);
+	
+		Direction = EyeRotation.Vector();
 		
 		FVector TraceStart = EyeLocation;
-		const float DistanceToPlayer = (PlayerOwner->GetActorLocation() - TraceStart).Size();
-		constexpr float ExtraDistance = 10.0f;
+		const float DistanceToPlayer = (OwnerCharacter->GetActorLocation() - TraceStart).Size();
+		constexpr float ExtraDistance = 100.0f;
```

Comment:
This change makes it seem like this should be a editable property

Reply:
![image](https://user-images.githubusercontent.com/85327912/173109227-2260b76c-d959-40f1-991a-c68e3bc381e9.png)